### PR TITLE
More TS

### PR DESCRIPTION
**Pull request by wagenet**

### CODE REVIEW

**Comment by wagenet** [comment]
This will need to be cleaned up before merging.

File: CHANGELOG.md
Comment (chriskrycho) [inline CHANGELOG.md:3147]: 😂 Love it.

File: packages/@ember/-internals/container/lib/registry.ts
Comment (chriskrycho) [reply packages/@ember/-internals/container/lib/registry.ts:84]: Noting how it works on line 64, are we *sure* it's not right? (It's certainly *weird*, and I don't know the runtime code here at all so I'll defer to you on the judgment call.)

https://github.com/emberjs/ember.js/blob/9b0d2ff6f831f0795f0ed5a142e149bfbc29ec23/packages/%40ember/-internals/container/lib/registry.ts#L64

File: packages/@ember/-internals/extension-support/lib/container_debug_adapter.ts
Comment (chriskrycho) [inline packages/@ember/-internals/extension-support/lib/container_debug_adapter.ts:65]: Given it's set in the constructor, shouldn't need the `declare` here.

```suggestion
  resolver: Resolver;
```

File: packages/@ember/-internals/runtime/lib/ext/rsvp.ts
Comment (chriskrycho) [inline packages/@ember/-internals/runtime/lib/ext/rsvp.ts:45]: Optional chaining not needed here given the `!reason` check on line 42.

```suggestion
  if (withErrorThrown.errorThrown) {
```

File: packages/@ember/-internals/runtime/lib/mixins/action_handler.d.ts
Comment (chriskrycho) [inline packages/@ember/-internals/runtime/lib/mixins/action_handler.d.ts:5]: Will presumably need a rebase now that #19948 has also updated this?

File: packages/@ember/-internals/runtime/lib/mixins/promise_proxy.d.ts
Comment (chriskrycho) [inline packages/@ember/-internals/runtime/lib/mixins/promise_proxy.d.ts:4]: The default value is presumably `null` but this would make it so you can't use anything *but* `null`. From the docs, I think it should be `unknown`?

```suggestion
  reason: unknown;
```

**Review by chriskrycho — CHANGES_REQUESTED** [review]
Couple small tweaks.

File: packages/@ember/-internals/container/lib/registry.ts
Comment (wagenet) [reply packages/@ember/-internals/container/lib/registry.ts:84]: Probably deserves a more careful look at least!

File: packages/@ember/-internals/runtime/lib/mixins/action_handler.d.ts
Comment (wagenet) [reply packages/@ember/-internals/runtime/lib/mixins/action_handler.d.ts:5]: Yes

**Review by chriskrycho — APPROVED** [review]

**Pull request merged**
